lib/hashie/extensions/coercion/active_model.rb

```diff
@@ -0,0 +1,85 @@
+require 'active_model/type'
+
+module Hashie
+  module Extensions
+    module Coercion
+      module ActiveModel
+        # Symbol is the one glaring type omission. Define it if it is not
+        # already defined.
+        begin
+          ::ActiveModel::Type.lookup(:symbol)
+        rescue ArgumentError
+          class Symbol < ::ActiveModel::Type::String
+            def type
+              :symbol
+            end
+
+            private
+
+            # The symbol type leverages in coercion logic in the String
+            # superclass and simply uses `to_sym` to covert to a symbol.
+            def cast_value(value)
+              super.to_sym
+            end
+          end
+          ::ActiveModel::Type.register(:symbol, Symbol)
+        end
+
+        # Array of symbols of the included ActiveModel types.
+        ACTIVE_MODEL_TYPES = ::ActiveModel::Type.registry
+                             .send(:registrations)
+                             .map { |r| r.send(:name) }
+                             .freeze
+
+        def build_coercion(type)
+          if type.is_a? Enumerable
+            if type.class <= ::Hash
+              type, key_type, value_type = type.class, *type.first
+              build_hash_coercion(type, key_type, value_type)
+            else # Enumerable but not Hash: Array, Set
+              value_type = type.first
+              type = type.class
+              build_container_coercion(type, value_type)
+            end
+          elsif ACTIVE_MODEL_TYPES.include? type
+            build_active_model_type_coercion(type)
+          elsif type.respond_to? :coerce
+            lambda do |value|
+              return value if value.is_a? type
+              type.coerce(value)
+            end
+          elsif type.respond_to? :new
+            lambda do |value|
```

Comment:
Same here. Do we need the lambda?

I can see the argument more for this one than for the `#coerce`, but I'm not sure what the user expects here. If you pass an object into a coercible hash, it would seem to me like you don't care about the original object anymore -- you should care about the resulting coercion.

With that logic, I think it should be safe just to call `type.new(value)` and skip the lambda. Do you agree?